Portal_Perfomance_Employes/Controllers/LoginController.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.IdentityModel.Tokens;
+using PortalPerfomanceEmployees.Data;
+using PortalPerfomanceEmployees.Models;
+
+namespace PortalPerfomanceEmployees.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class LoginController : ControllerBase
+    {
+        private readonly AppDbContext _context;
+        private readonly IConfiguration _config;
+
+        public LoginController(AppDbContext context, IConfiguration config)
+        {
+            _context = context;
+            _config = config;
+        }
+
+        [AllowAnonymous]
+        [HttpPost]
+        public IActionResult Login(UserDTO userLogin)
+        {
+            var user = Autenticate(userLogin);
+            if (user != null)
+            {
+                string token = GenerateToken(user);
+                return Ok(token);
+            }
+
+            return NotFound("User not found.");
+        }
+
+        private string GenerateToken(User user)
+        {
+            var securityKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(_config["Jwt:Key"]));
+            var credentials = new SigningCredentials(securityKey, SecurityAlgorithms.HmacSha256);
+
+            var claims = new[]
+            {
+                new Claim(ClaimTypes.NameIdentifier, user.Username),
+                new Claim(ClaimTypes.Email, user.EmailAddress),
+                new Claim(ClaimTypes.GivenName, user.GivenName),
+                new Claim(ClaimTypes.Role, user.Role.ToString())
+            };
+
+            var token = new JwtSecurityToken(_config["Jwt:Issuer"],
+              _config["Jwt:Audience"],
+              claims,
+              expires: DateTime.Now.AddMinutes(30),
+              signingCredentials: credentials);
+
+            return new JwtSecurityTokenHandler().WriteToken(token);
+
```

Comment:
spaces

Reply:
Should i create one role on .net analyzer about it?

Reply:
removed the empty line